casanovo/denovo/model.py

```diff
@@ -992,147 +991,123 @@ def configure_optimizers(
 
 class DbSpec2Pep(Spec2Pep):
     """
-    Inherits Spec2Pep
+    Subclass of Spec2Pep for the use of Casanovo as an \
+    MS/MS database search score function.
 
-    Hijacks teacher-forcing implemented in Spec2Pep and
-    uses it to predict scores between a spectra and associated peptide.
+    Uses teacher forcing to 'query' Casanovo for its score for each AA
+    within a candidate peptide, and takes the geometric average of these scores
+    and reports this as the score for the spectrum-peptide pair. Note that the
+    geometric mean of the AA scores is actually calculated by a
+    summation and average of the log of the scores, to preserve numerical
+    stability. This does not affect PSM ranking.
+
+    Also note that although teacher-forcing is used within this method,
+    there is *no training* involved. This is a prediction-only method.
+
+    Output is provided in .mztab format.
     """
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
+        self.psm_batch_size = 1024
 
     def predict_step(self, batch, *args):
         """
         A single prediction step for Casanovo-DB
 
         Parameters
         ----------
-        batch : Tuple[torch.Tensor, torch.Tensor, torch.Tensor]
+        batch : Tuple[torch.Tensor, torch.Tensor, np.array, List[str], List[str]]
             A batch of (i) MS/MS spectra, (ii) precursor information, (iii)
-            spectrum identifiers as torch Tensors, (iv) scan numbers.
+            spectrum identifiers, (iv) candidate peptides, (v) associated proteins.
```

Comment:
**question:** Can we avoid passing the associated proteins through the whole predicting step and instead just use peptides, and then afterwards, when the output results are compiled, map the peptides back to the proteins? That could be handled in `ProteinDatabase`. Benefit would be to avoid requiring all this memory for all candidates during inference.

Reply:
Removed passing of proteins through all levels